### PR TITLE
remove batch response

### LINKED ISSUE #1019: What is the point of BatchResponse?
I am working on #987 and reading the batch handling related code and just cannot make sense of `BatchResponse`

https://github.com/paritytech/jsonrpsee/blob/b98f44369f045a0f6fde5abdb3cbbcebfb830bdc/core/src/server/helpers.rs#L346-L353

The `success` is only used in tests, which can inspect the result to check error or not.

Maybe I should refactor it out before making more changes to batch request handling code?

### PR DESCRIPTION
**Pull request by xlc**
closes #1019

### CODE REVIEW

File: core/src/server/helpers.rs [hunk 1]
```diff
@@ -332,32 +332,20 @@ impl BatchResponseBuilder {
...
-
-impl BatchResponse {
-	/// Create a `BatchResponse` from an error.
-	pub fn error(id: Id, err: impl Into<ErrorObject<'static>>) -> Self {
-		let result = serde_json::to_string(&ErrorResponse::borrowed(err.into(), id)).unwrap();
-		Self { result, success: false }
-	}
+/// Create a JSON-RPC error response.
+pub fn batch_response_error(id: Id, err: impl Into<ErrorObject<'static>>) -> String {
+	serde_json::to_string(&ErrorResponse::borrowed(err.into(), id)).unwrap()
```
Comment (niklasad1) [inline core/src/server/helpers.rs:348]:
```suggestion
	serde_json::to_string(&ErrorResponse::borrowed(err.into(), id)).expect("ErrorResponse Serialize is infallible; qed");
```

**Review by niklasad1 — APPROVED** [review]
LGTM

**Comment by xlc** [comment]
@niklasad1 good to merge this?

**Comment by niklasad1** [comment]
it just needs another review

**Review by lexnv — APPROVED** [review]

**Pull request merged**